go/test/endtoend/vtcombo/recreate/recreate_test.go

```diff
@@ -101,13 +101,36 @@ func TestDropAndRecreateWithSameShards(t *testing.T) {
 
 	cur := conn.Session(ks1+"@primary", nil)
 
+	mysqlConnCountBefore, err := getMySQLConnectionCount(ctx, cur)
+	require.Nil(t, err)
+
 	_, err = cur.Execute(ctx, "DROP DATABASE "+ks1, nil)
 	require.Nil(t, err)
 
 	_, err = cur.Execute(ctx, "CREATE DATABASE "+ks1, nil)
 	require.Nil(t, err)
 
 	assertTabletsPresent(t)
+
+	mysqlConnCountAfter, err := getMySQLConnectionCount(ctx, cur)
+	require.Nil(t, err)
```

Comment:
Why not test it after the `DROP` and before the `CREATE`? Isn't that where the problem was?

Reply:
There are zero connections after the `DROP`. I'm assuming the existing connections get closed and cannot be reestablished by the pools. After the `CREATE` the pools reestablish connections and that's when the leak becomes apparent.

Reply:
OK, cool. That's worth a comment then in the test IMO.